lib/monitor/stats.go

```diff
@@ -276,13 +202,90 @@ func (df *DataFilter) Filter(measure string, values []float64) []float64 {
 	return values[:maxIndex]
 }
 
-// value is used to compute the statistics
+// Collect make the final computations before stringing or writing.
+// Autmatically done in other methods anyway.
+func (s *Stats) Collect() {
+	for _, v := range s.values {
+		v.Filter(s.filter)
+		v.Collect()
+	}
+}
+
+func (s *Stats) Value(name string) *Value {
+	if val, ok := s.values[name]; ok {
+		return val
+	}
+	return nil
+}
+
+// Returns an overview of the stats - not complete data returned!
+func (s *Stats) String() string {
+	var str string
+	for _, k := range s.staticKeys {
+		str += fmt.Sprintf("%s = %d", k, s.static[k])
+	}
+	for _, v := range s.values {
+		str += fmt.Sprintf("%v", v.Values())
+	}
+	return fmt.Sprintf("{Stats: %s}", str)
+}
+
+// Read a config file and fills up some fields for Stats struct
+func (s *Stats) readRunConfig(rc map[string]string, defaults ...string) {
+	// First find the defaults keys
+	for _, def := range defaults {
+		valStr, ok := rc[def]
+		if !ok {
+			dbg.Fatal("Could not find the default value", def, "in the RunConfig")
+		}
+		if i, err := strconv.Atoi(valStr); err != nil {
+			dbg.Fatal("Could not parse to integer value", def)
+		} else {
+			// registers the static value
+			s.static[def] = i
+			s.staticKeys = append(s.staticKeys, def)
+		}
+	}
+	// Then parse the others keys
+	statics := make([]string, 0)
+	for k, v := range rc {
+		// pass the ones we already registered
+		var alreadyRegistered bool
+		for _, def := range defaults {
+			if k == def {
+				alreadyRegistered = true
+				break
+			}
+		}
+		if alreadyRegistered {
+			continue
+		}
+		// store it
+		if i, err := strconv.Atoi(v); err != nil {
+			dbg.Error("Could not parse the value", k, "from runconfig (v=", v, ")")
```

Comment:
There are also string-values in the config-file. I propose to change the `dbg.Error` into `dbg.Lvl2` or even `dbg.Lvl3`, as it will 'error' on nearly any .toml-file.